examples/squiral/main.go

```diff
@@ -0,0 +1,371 @@
+			}
+			if auto.colorMap[ntarg.x][ntarg.y] == s.col {
+				// If this has the same color, we cannot go into this direction,
+				// to avoid ugly blocks of equal color.
+				continue // try next direction
+			}
+
+			// b. Test all outer fields for the color of the
+			// squiral itself.
+			horivert := dir % 2
+			xtarg := tup{}
+			set := true
+			for _, out := range outers[horivert] {
+				xtarg.x = target.x + out.x
+				xtarg.y = target.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+
+				xtarg.x = ntarg.x + out.x
+				xtarg.y = ntarg.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+			}
+
+			if set {
+				s.pos = target
+				s.dir = dir
+				// 2. set the color of this squiral to its
+				// current position.
+				setpix(s.pos, s.col)
+				return
+			}
+		}
+	}
+
+	s.dead = true
+}
+
+type automaton struct {
+	squirals [amount]squiral
+	colorMap [width][height]color.Color
+}
+
+func (au *automaton) init() {
+	// Init the test grid with color (0,0,0,0) and the borders of
+	// it with color(0,0,0,254) as a blocker color, so the squirals
+	// cannot escape the scene.
+	for x := 0; x < width; x++ {
+		for y := 0; y < height; y++ {
+			if x == 0 || x == width-1 || y == 0 || y == height-1 {
+				auto.colorMap[x][y] = blocker
+			} else {
+				auto.colorMap[x][y] = free
+			}
+		}
+	}
+
+	for i := 0; i < amount; i++ {
+		auto.squirals[i].spawn()
+	}
+}
+
+func (au *automaton) step() {
+	for i := 0; i < amount; i++ {
+		for s := 0; s < au.squirals[i].speed; s++ {
+			au.squirals[i].step(i)
+			if au.squirals[i].dead {
+				au.squirals[i].spawn()
+			}
+		}
+	}
+}
+
+func setpix(xy tup, col color.Color) {
+	canvas.Set(xy.x, xy.y, col)
+	auto.colorMap[xy.x][xy.y] = col
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	c, err := ebiten.NewImage(width, height, ebiten.FilterNearest)
+	if err != nil {
+		panic(err)
+	}
+	canvas = c
+	canvas.Fill(colornames.Black)
+
+	auto = &automaton{}
+	auto.init()
+}
+
+func update(screen *ebiten.Image) error {
+	reset := false
+
+	if ebiten.IsKeyPressed(ebiten.KeyEscape) {
+		return quit
+	}
+
+	if ebiten.IsKeyPressed(ebiten.KeyB) {
+		if background == colornames.White {
+			background = colornames.Black
+		} else {
+			background = colornames.White
+		}
+		reset = true
+	} else if ebiten.IsKeyPressed(ebiten.KeyT) {
+		selectedPalette = (selectedPalette + 1) % len(palettes)
+		reset = true
+	}
+
+	if ebiten.IsMouseButtonPressed(ebiten.MouseButtonLeft) {
+		reset = true
+	}
+
+	if reset {
+		screen.Fill(background)
```

Comment:
I don't think we need to fill `screen` since the screen will be rendered with `canvas` anytime.